packages/augur-core/src/contracts/trading/ERC20Proxy.sol

```diff
@@ -0,0 +1,126 @@
+    mapping (uint256 => mapping (address => mapping (address => uint256))) public allowances;
+    // tokenId -> proxy contract
+    mapping (uint256 => address) public proxies;
+
+    constructor(address _proxyToClone, address _target1155) public {
+        proxyToClone = _proxyToClone;
+        target1155 = _target1155;
+        // This contract will be able to do anything with the user's 1155 tokens.
+        IERC1155(target1155).setApprovalForAll(address(this), true);
+    }
+
+    function newERC20(uint256 _tokenId) public returns (ERC20Proxy) {
+        require(proxies[_tokenId] == address(0)); // must not have been registered yet
+        _erc20 = ERC20Proxy(createClone(proxyToClone));
+        proxies[_tokenId] = _erc20; // register
+        _erc20.initialize(this, _tokenId);
+        return _erc20;
+    }
+
+    function transfer(address _proxy, address _caller, address _to, uint256 _amount) public returns (bool) {
+        _transfer(_proxy, _caller, _to, _amount);
+        return true;
+    }
+
+    function transferFrom(address _proxy, address _caller, address _from, address _to, uint256 _amount) public returns (bool) {
+        _transfer(_proxy, _from, _to, _amount);
+        _approve(_proxy, _from, _caller, allowances[_tokenId][_from][_caller].sub(_amount));
+        return true;
+    }
+
+    function approve(address _proxy, address _caller, address _spender, uint256 _amount) public returns (bool) {
+        _approve(_proxy, _caller, _spender, _amount);
+        return true;
+    }
+
+    function allowance(address _proxy, address owner, address spender) public returns (uint256) {
+        uint256 _tokenId = _gate(_proxy);
+        return allowances[_tokenId][_owner][_spender];
+    }
+
+    function _transfer(address _proxy, address _from, address _to, uint256 _amount) internal {
+        ERC20Proxy _proxyContract = ERC20Proxy(_proxy);
+        uint256 _tokenId = _proxyContract.tokenId();
+        _gate(_proxyContract, _tokenId);
+
+        IERC1155(target1155).safeTransferFrom(_from, _to, _tokenId, _amount, bytes(0));
+        _proxyContract.logTransfer(_from, _to, _amount);
+    }
+
+    function _approve(address _proxy, address _owner, address _spender, uint256 _amount) internal {
+        ERC20Proxy _proxyContract = ERC20Proxy(_proxy);
+        uint256 _tokenId = _proxyContract.tokenId();
+        _gate(_proxyContract, _tokenId);
+
+        allowances[_tokenId][_owner][_spender] = _amount;
+        _proxyContract.logApproval(_owner, _spender, _amount);
+    }
+
+    // Guarantees that the calling function was invoked by a trusted proxy contract.
+    // So we can trust its _caller to be the proxy's msg.sender.
+    function _gate(ERC20Proxy _proxy, uin256 _tokenId) internal {
+        require(msg.caller == _proxy); // only proxies may make calls
+        require(proxies[_tokenId] == _proxy); // no sneaky stuff!
+    }
+}
+
+contract ERC20Proxy is IERC20 {
+    uint8 constant public decimals = 18;
```

Comment:
In general we just use uint256 for all integer types. Easier to audit / some auditors have recommended this practice.

Reply:
Our ERC20 contract uses uint8 here.